targets/utils.py

```diff
@@ -139,6 +139,14 @@ def platform_toolchain_extend(platform, cmdname, argument):
     bt.clear()
     bt.extend(_platform_toolchain_cmd_join(cmds))
 
+def define_flash_constants(soc):
+    soc.add_constant("FLASH_BOOT_ADDRESS", soc.flash_boot_address)
+    if soc.cpu_variant == "linux":
```

Comment:
BTW It would be much better for these to be ordered,

 - Emulator
 - DTS
 - Kernel
 - Root FS

Emulator + DTS could be needed for non-Linux applications / operating systems. Both should be pretty small / constant size.

The RootFS is the one which is most likely to be variable in size so should be at the one at end.

Probably should also add constants for `MAX_SIZE` here?